src/__experimental__/components/date/date.component.js

```diff
@@ -0,0 +1,242 @@
+  };
+
+  handleBlur = (ev) => {
+    const { disabled, readOnly } = this.props;
+
+    if (disabled || readOnly || this.isBlurBlocked) return;
+
+    if (this.props.onBlur) this.props.onBlur(ev);
+  }
+
+  handleFocus = (ev) => {
+    const { disabled, readOnly } = this.props;
+
+    if (disabled || readOnly) return;
+
+    if (this.isAutoFocused) {
+      this.isAutoFocused = false;
+    } else {
+      this.openDatePicker();
+    }
+
+    if (this.props.onFocus) this.props.onFocus(ev);
+  };
+
+  handleKeyDown = (ev) => {
+    if (Events.isTabKey(ev)) {
+      this.closeDatePicker();
+    }
+  };
+
+  openDatePicker = () => {
+    this.isBlurBlocked = true;
+    document.addEventListener('click', this.closeDatePicker);
+    this.updateDatePickerValue(this.state.visibleValue);
+    this.setState({ isDatePickerOpen: true });
+  };
+
+  closeDatePicker = () => {
+    document.removeEventListener('click', this.closeDatePicker);
+    this.setState((prevState) => {
+      return {
+        visibleValue: formatVisibleValue(prevState.selectedDate),
+        isDatePickerOpen: false
+      };
+    });
+  };
+
+  handleDateSelect = (selectedDate) => {
+    const stringDateIso = DateHelper.formatDateString(selectedDate, isoDateFormat);
+
+    this.isBlurBlocked = true;
+    this.closeDatePicker();
+    this.updateVisibleValue(selectedDate);
+    this.emitOnChangeCallback(stringDateIso);
+  };
+
+  updateVisibleValue = (date) => {
+    const formattedDateString = formatVisibleValue(date);
+    this.setState({ visibleValue: formattedDateString });
+  };
+
+  handleVisibleInputChange = (ev) => {
+    const { disabled, readOnly } = this.props;
+    const dateWithSlashes = DateHelper.sanitizeDateInput(ev.target.value);
+    const isValidDate = DateHelper.isValidDate(dateWithSlashes);
+    const isoDateString = DateHelper.formatDateString(dateWithSlashes, isoDateFormat);
+
+    if (disabled || readOnly) return;
+
+    // Updates the hidden value after first formatting to default hidden format
+    if (isValidDate) {
+      this.updateDatePickerValue(dateWithSlashes);
+      this.emitOnChangeCallback(isoDateString);
+    }
+
+    this.setState({ visibleValue: ev.target.value });
+  };
+
+  updateDatePickerValue = (newValue) => {
+    const isoDate = convertToIsoDate(newValue);
+
+    this.setState({ selectedDate: isoDate });
+  };
+
+  emitOnChangeCallback = (stringDate) => {
+    const changePayload = { target: { value: stringDate } };
+
+    if (this.props.onChange) this.props.onChange(changePayload);
+  };
+
+  renderDatePicker = (dateRangeProps) => {
+    const datePickerProps = {
+      inputElement: this.input && this.input.parentElement,
+      selectedDate: this.state.selectedDate,
+      handleDateSelect: this.handleDateSelect,
+      ...dateRangeProps
+    };
+
+    return <DatePicker { ...datePickerProps } />;
+  }
+
+  render() {
+    const { minDate, maxDate, ...inputProps } = this.props;
+    let events = {};
+    delete inputProps.autoFocus;
```

Comment:
Seems like a lot of messing around with this `autoFocus` just for it to get deleted on the first render? Maybe I'm missing the point?

Reply:
these props are passed down to the `Textbox` component, we don't need that property there as we are using it directly in the `Date` component